tree/treeplayer/inc/ROOT/TDFInterface.hxx

```diff
@@ -364,6 +364,27 @@ public:
       TInterface<TTraits::TakeFirstParameter_t<decltype(upcastNode)>> upcastInterface(fProxiedPtr, fImplWeakPtr);
       // build a string equivalent to
       // "(TInterface<nodetype*>*)(this)->Snapshot<Ts...>(treename,filename,*(ColumnNames_t*)(&columnList))"
+
+      // The interpretation has multiple steps including
+      // 1. Parse
+      // 2. Codegen
+      // 3. Link
+      // 4. run static initializers
+      // 5. run user code
+      // 6. run static destructors.
+      //
+      // Where 1,2,3 explicitly modify the state of the (global) interpreter and must be (and, of course, are)
+      // covered by the ROOT global lock.  Technically, cling should release the lock when executing user
+      // code at step 4,5,6 but currently does not.  Since the user code can create threads that may
+      // want to get the lock (to access the global data in either Core or Clang), not releasing the lock
+      // can lead to deadlock .. hence the 'gROOTMutex->UnLock' belows ... Once Cling is updated to
+      // release the lock, this line (and the lock taking later on) must be removed.
+      //
+      // Updating Cling properly has challenges as step 4,5,6 may have implied steps like
```

Comment:
While that's correct, this is probably not the place to discuss cling implementation issues?